core/station/impl/src/controllers/http.rs

```diff
@@ -2,12 +2,15 @@ use crate::{core::ic_cdk::api::canister_balance, SERVICE_NAME};
 use ic_cdk_macros::query;
 use lazy_static::lazy_static;
 use orbit_essentials::api::{HeaderField, HttpRequest, HttpResponse};
+use orbit_essentials::http::add_skip_certification_headers;
 use orbit_essentials::metrics::with_metrics_registry;
 
 // Canister entrypoints for the controller.
 #[query(name = "http_request", decoding_quota = 10000)]
 async fn http_request(request: HttpRequest) -> HttpResponse {
-    CONTROLLER.router(request).await
+    let mut resp = CONTROLLER.router(request).await;
+    add_skip_certification_headers(&mut resp);
```

Comment:
this is failing is tests because it's calling the actual `set_certified_data` which is only available in a canister environment, and using `orbit_essentials::http::add_skip_certification_headers` the #[cfg(test)] does not apply across dependencies. for this to work you'll likely need to pass the `set_certified_data` as a callback and get that by importing in the control-panel and station with `crate::core::ic_cdk::api::set_certified_data`

Reply:
Let me then just return the certified data to be set from the library and invoke the system API here.